### ISSUE TITLE
MinigetError: Status code: 410

### ISSUE THREAD

**Issue by erick-gomes** [fent/node-ytdl-core#985]
MinigetError: Status code: 410
     at ClientRequest.<anonymous> (/app/node_modules/miniget/src/index.ts:257:19)
     at Object.onceWrapper (events.js:520:26)
     at ClientRequest.emit (events.js:400:28)
     at ClientRequest.emit (domain.js:470:12)
     at HTTPParser.parserOnIncomingClient (_http_client.js:647:27)
     at HTTPParser.parserOnHeadersComplete (_http_common.js:126:17)
     at TLSSocket.socketOnData (_http_client.js:515:22)
     at TLSSocket.emit (events.js:400:28)
     at TLSSocket.emit (domain.js:470:12)
     at addChunk (internal/streams/readable.js:290:12) {
    statusCode: 410
 }

**Comment by Nico-ITA** [comment]
Its seems that this error is similar too 404 error so you can simple try to use cookie . Also try to play this video https://www.youtube.com/watch?v=dmJefsOErr0 (is a age restricted video ) and reply here the results

**Comment by erick-gomes** [comment]
```typescript

import { getURLVideoID, getInfo, validateURL, downloadFromInfo } from 'ytdl-core'
import yts from 'yt-search'
/*
    Age restriction information is reported only after the error, if the restriction is active, I will never know that the error was 
    caused by the restriction.
*/
(async () => {
   try {
          const video = (await yts(search)).videos.slice(0, 3)[0]
          const videoId = getURLVideoID('https://www.youtube.com/watch?v=dmJefsOErr0') // restricted video
          const info = await getInfo(videoId) // The error happens on this line due to the age restriction
          if (video.seconds > 300) {
              msg.reply('Escolha outro vídeo, esse é muito grande.')
              return
          }
          
          // this alert message about the age restriction
          info.videoDetails.age_restricted && msg.reply('⚠️Aviso! Esse vídeo tem restrição de idade!') 
          
          
          console.log(info)
    } catch (err) {
        console.log(err) // output MinigetError Status code 410
    }
})()



// https://www.youtube.com/watch?v=YVJijQIualA , an unrestricted video has the normal output like this:


author: {
      id: 'UCxVxcprt9bqQQ-95sMLLRoQ',
      name: 'Midas Music',
      user: 'MidasBrasil',
      channel_url: 'https://www.youtube.com/channel/UCxVxcprt9bqQQ-95sMLLRoQ',
      external_channel_url: 'https://www.youtube.com/channel/UCxVxcprt9bqQQ-95sMLLRoQ',
      user_url: 'http://www.youtube.com/user/MidasBrasil',
      thumbnails: [Array],
      verified: false,
      subscriber_count: 3140000
    },
    isPrivate: false,
    isUnpluggedCorpus: false,
    isLiveContent: false,
    media: {
      song: 'O Sol',
      category: 'Music',
      category_url: 'https://music.youtube.com/',
      artist: 'Vitor Kley',
      'licensed to youtube by': 'UMPI, UNIAO BRASILEIRA DE EDITORAS DE MUSICA - UBEM, LatinAutorPerf, UMPG Publishing, BMI - Broadcast Music Inc., LatinAutor - UMPG, and 6 Music Rights Societies'
    },
    likes: 2527089,
    dislikes: 86030,
    age_restricted: false,
    video_url: 'https://www.youtube.com/watch?v=YVJijQIualA',
    storyboards: [ [Object], [Object], [Object] ],
    chapters: [],
    thumbnails: [ [Object], [Object], [Object], [Object], [Object] ]
  },
  
```

**Comment by Nico-ITA** [comment]
Try to use cookie https://github.com/fent/node-ytdl-core/blob/master/example/cookies.js ( here official example ).

Here an community example https://github.com/fent/node-ytdl-core/issues/980#issuecomment-886211227

**Comment by r0hin** [comment]
Having the same issue. Happens with specific videos on certain devices. IE, locally, it's fine but on a Google Cloud Run instance, throws the error.

**Comment by ProGamer2711** [comment]
I have the same issue. When I have to play an age restricted video my discord bot crashes. tried the cookie example but it still won't work. Can anybody please help.

**Comment by ProGamer2711** [comment]
> 
> 
> ```ts
> import { getURLVideoID, getInfo, validateURL, downloadFromInfo } from 'ytdl-core'
> import yts from 'yt-search'
> /*
>     Age restriction information is reported only after the error, if the restriction is active, I will never know that the error was 
>     caused by the restriction.
> */
> (async () => {
>    try {
>           const video = (await yts(search)).videos.slice(0, 3)[0]
>           const videoId = getURLVideoID('https://www.youtube.com/watch?v=dmJefsOErr0') // restricted video
>           const info = await getInfo(videoId) // The error happens on this line due to the age restriction
>           if (video.seconds > 300) {
>               msg.reply('Escolha outro vídeo, esse é muito grande.')
>               return
>           }
>           
>           // this alert message about the age restriction
>           info.videoDetails.age_restricted && msg.reply('⚠️Aviso! Esse vídeo tem restrição de idade!') 
>           
>           
>           console.log(info)
>     } catch (err) {
>         console.log(err) // output MinigetError Status code 410
>     }
> })()
> 
> 
> 
> // https://www.youtube.com/watch?v=YVJijQIualA , an unrestricted video has the normal output like this:
> 
> 
> author: {
>       id: 'UCxVxcprt9bqQQ-95sMLLRoQ',
>       name: 'Midas Music',
>       user: 'MidasBrasil',
>       channel_url: 'https://www.youtube.com/channel/UCxVxcprt9bqQQ-95sMLLRoQ',
>       external_channel_url: 'https://www.youtube.com/channel/UCxVxcprt9bqQQ-95sMLLRoQ',
>       user_url: 'http://www.youtube.com/user/MidasBrasil',
>       thumbnails: [Array],
>       verified: false,
>       subscriber_count: 3140000
>     },
>     isPrivate: false,
>     isUnpluggedCorpus: false,
>     isLiveContent: false,
>     media: {
>       song: 'O Sol',
>       category: 'Music',
>       category_url: 'https://music.youtube.com/',
>       artist: 'Vitor Kley',
>       'licensed to youtube by': 'UMPI, UNIAO BRASILEIRA DE EDITORAS DE MUSICA - UBEM, LatinAutorPerf, UMPG Publishing, BMI - Broadcast Music Inc., LatinAutor - UMPG, and 6 Music Rights Societies'
>     },
>     likes: 2527089,
>     dislikes: 86030,
>     age_restricted: false,
>     video_url: 'https://www.youtube.com/watch?v=YVJijQIualA',
>     storyboards: [ [Object], [Object], [Object] ],
>     chapters: [],
>     thumbnails: [ [Object], [Object], [Object], [Object], [Object] ]
>   },
>   
> ```
Hello I wonder is it necessary for the youtube alts to be verified and also how do I find the headers that I'm searching for because I'm having problems with that. Thanks in advance.

**Comment by GabFLAB** [comment]
All the videos are returning me this error

**Comment by face-hh** [comment]
same here, with the example from README.md

**Comment by SetaWinsU** [comment]
same here, any solution?

**Comment by Claxtastic** [comment]
also getting this error for many videos. Restarting my app will often resolve the issue, but I'm not sure what to do other than that.

**Comment by lollipop-0** [comment]
I also have the problem with age-restricted videos with the _.getBasicInfo()_ function. The problem is, there is no way to check if they are age restricted because checking them will get you the error. 

You can prevent the crash with a _try ... catch_ statement.

**Comment by PhanNN** [comment]
If you guys just want to download the mp3 file only. 
I suggest you should use a fallback solution if the node-ytdl doesn't work
I try to use "https://www.yt-download.org/developers", get the mp3 links from html response and download by miniget (or another lib)

Good luck!

**Comment by r0hin** [comment]
> If you guys just want to download the mp3 file only.
> I suggest you should use a fallback solution if node-ytdl doesn't work
> I try to use "https://www.yt-download.org/developers", get the mp3 links from html response and download by miniget (or another lib)
> 
> Good luck!

For anyone who needs a simple implementation of the above fallback solution, here is mine:
```
response = await fetch(`https://www.yt-download.org/api/button/mp3/${queryURL.split('?v=')[1]}`);
text = await response.text();
url = backupResult.split('<a href="')[1].split('" class="shadow-xl')[0];```

**Comment by basserd** [comment]
Still having this error. Seems to be kinda random for me, restarting the application often helps. I also have the error with the age restricted video.

**Comment by sayho93** [comment]
I'm getting this error when playing Youtube-music video. Also, inside ytdl-core/index.js youtube returns status: 'UNPLAYABLE',
      reason: 'Video unavailable'
any updates on this problem?

**Comment by nikolaigeorgie** [comment]
Getting this issue randomly in my ec2 instance with aws.. no cookies or headers.

**Comment by leon20121005** [comment]
Also getting this issue for all videos on Heroku.
` const information = await ytdl.getInfo(url); `
![image](https://user-images.githubusercontent.com/19206746/136681061-16051142-f65d-44f4-a4d7-dd4ad9ca5e12.png)

**Comment by nikolaigeorgie** [comment]
@leon20121005 i switched my queue to a new redis instance and that got it working. seems like after a certain amount of requests could be youtube is rate limiting..

**Comment by nikolaigeorgie** [comment]
> switched to https://www.npmjs.com/package/youtube-dl-exec ... good alternative

youtube-dl-exec package behaves much slower than this package.. *sigh* , oh miniget ... lol

**Comment by nicocespede** [comment]
> @nikolaigeorgie Well... I just added error handlings without changing its functionality and got it working too lol.

Can you tell me how you handled the errors please? I started getting this error last night and I can't fix it, I'm hosting my bot on Heroku

**Comment by nikolaigeorgie** [comment]
@nicocespede i have no solution to this issue yet. i've switched to https://www.npmjs.com/package/youtube-dl-exec for now

**Comment by TimeForANinja** [comment]
should be fixed with #1022
